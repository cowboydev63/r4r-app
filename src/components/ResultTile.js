import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';
import { Link } from 'react-router-dom';

class ResultTile extends React.PureComponent {
    static propTypes = {
        title: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
    }

    static defaultProps = {
        title: '',
        description: '',
        id: '',
    }

    render() {
        return (
            <Theme element="article" className="result-tile" aria-label="search result">
                <h2><Link to={`/resource/${ this.props.id }`}>{ this.props.title }</Link></h2>
                <p dangerouslySetInnerHTML={{ __html: this.props.description }}/>
            </Theme>
        )
    }
}

export default ResultTile;