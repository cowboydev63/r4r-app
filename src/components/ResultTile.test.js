import React from 'react';
import { shallow } from 'enzyme';
import ResultTile from './ResultTile';

describe('ResultTile Component', () => {
  it('renders without crashing', () => {
    shallow(<ResultTile id={1} />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<ResultTile id={1} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('appends a click handler to the result link', () => {
    const fn = jest.fn()
    const wrapper = shallow(<ResultTile id={1} onClick={fn}/>);
    wrapper.find('Link').simulate('click');
    expect(fn.mock.calls.length).toBe(1);
  })
})