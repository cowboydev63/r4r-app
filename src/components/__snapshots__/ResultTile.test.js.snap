// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultTile Component renders correctly 1`] = `
<Theme
  aria-label="search result"
  className="result-tile"
  element="article"
>
  <h2>
    <Link
      onClick={[Function]}
      replace={false}
      to="/resource/1"
    />
  </h2>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</Theme>
`;
